pkg/oauthserver/server/selectprovider/bootstrapselectprovider.go

```diff
@@ -25,7 +25,8 @@ type bootstrapSelectProvider struct {
 
 func (b *bootstrapSelectProvider) SelectAuthentication(providers []api.ProviderInfo, w http.ResponseWriter, req *http.Request) (*api.ProviderInfo, bool, error) {
 	// this should never happen but let us not panic the server in case we screwed up
-	if len(providers) == 0 || providers[0].Name != bootstrap.BootstrapUser {
+	// also avoids checking the secret when there is only one provider
+	if len(providers) <= 1 || providers[0].Name != bootstrap.BootstrapUser {
```

Comment:
I don't understand the one provider case. Can't the one provider be the BootstrapUser provider, in which case we'd want to validate the secret?

Reply:
This is basically "no need to filter a list of one because the OAuth server will always pick the one provider over returning a 500" - so even if I filtered this list down to zero, the OAuth server would still pick the one provider from the original list.  I can remove this later if we think it is pointless optimization.